File: frequency-of-the-most-frequent-element.cc
// Frequency of the Most Frequent Element
#define FOR(i, a, b) for (remove_cv<remove_reference<decltype(b)>::type>::type i = (a); i < (b); i++)
#define REP(i, n) FOR(i, 0, n)

class Solution {
public:
  int maxFrequency(vector<int>& a, int k) {
    long n = a.size(), j = 0, ans = 0, s = k;
    sort(a.begin(), a.end());
    REP(i, n) {
      while ((i-j)*a[i] > s)
        s -= a[j++];
      ans = max(ans, i-j+1);
      s += a[i];
    }
    return ans;
  }
};

File: frequency-of-the-most-frequent-element.rs
impl Solution {
    pub fn max_frequency(mut nums: Vec<i32>, k: i32) -> i32 {
        let mut k = k as i64;
        let mut j = 0;
        let mut ans = 0;
        nums.sort();
        for (i, x) in nums.iter().enumerate() {
            while (i-j) as i64 * *x as i64 > k {
                k -= nums[j] as i64;
                j += 1;
            }
            ans = std::cmp::max(ans, i-j+1);
            k += *x as i64;
        }
        ans as i32
    }
}
